velox/exec/tests/HashJoinTest.cpp

```diff
@@ -206,6 +206,44 @@ TEST_F(HashJoinTest, filter) {
       "((t_k0 % 100) + (u_k0 % 100)) % 40 < 20");
 }
 
+TEST_F(HashJoinTest, joinSidesDifferentSchema) {
+  // In this join, the tables have different schema. The filter predicate uses
+  // a column from the right table before the left and the corresponding
+  // columns at the same channel number(1) have different types. This has been
+  // a source of errors in the join logic.
+
+  auto leftType = makeRowType({BIGINT(), VARCHAR(), BIGINT()}, "t_");
+  auto rightType = makeRowType({BIGINT(), REAL(), BIGINT()}, "u_");
+
+  auto leftBatch = std::dynamic_pointer_cast<RowVector>(
+      BatchMaker::createBatch(leftType, 100, *pool_));
```

Comment:
BatchMaker is deprecated. Consider using VectorMaker instead. Checkout makeFlatVector methods in the base class.

Can you make sure the query returns non-empty results? This would be easier if you switch to VectorMaker.

Reply:
Alright. Yes, the VectorMaker apis give more control of the values generated. Have modified the test to use it.